docs/pages/dsDisplay/ds-display_L3_Test-Procedure.md

```diff
@@ -0,0 +1,305 @@
+
+## Setting Up Test Environment
+
+To execute `HAL` `L3` Python test cases, need a Python environment. Follow these steps mentioned in [HPK Public Documentation](https://github.com/rdkcentral/rdk-hpk-documentation/blob/main/README.md)
+
+### Update Configuration Files
+
+#### Rack Configuration File
+
+Example Rack configuration File: `ut/host/tests/configs/example_rack_config.yml`
+
+For more details refer [RAFT](https://github.com/rdkcentral/python_raft/blob/1.0.0/README.md) and [example_rack_config.yml](https://github.com/rdkcentral/python_raft/blob/1.0.0/examples/configs/example_rack_config.yml)
+
+In this file, update the configuration to define the console sessions for the `DUT` and the outbound settings:
+
+|Console Session|Description|
+|---------------|-----------|
+|default | Downloads the streams required for test cases.|
+|ssh_hal_test | Executes the HAL binary for the test case.|
+
+ ```yaml
+rackConfig:
+  - dut:
+      ip: "XXX.XXX.XXX.XXX"  # IP Address of the device
+      description: "stb device under test"
+      platform: "stb"
+      consoles:
+        - default:
+            type: "ssh"
+            port: 10022
+            username: "root"
+            ip: "XXX.XXX.XXX" # IP address of the device
+            password: ' '
+        - ssh_hal_test:
+            type: "ssh"
+            port: 10022
+            username: "root"
+            ip: "XXX.XXX.XXX" # IP address of the device
+            password: ' '
+      outbound:
+        download_url: "tftp://tftp-server.com/rack1/slot1/"    # Download location for the CPE device
+        upload_url: "sftp://server-address/home/workspace/tftp/rack1/slot1/" # Upload location
+        upload_url_base_dir: "sftp://server-address/home/workspace/tftp/rack1/slot1"
+        httpProxy:   # Local proxy if required
+        workspaceDirectory: './logs/workspace'   # Local working directory
+
+```
+
+#### Device Configuration File
+
+  Example Device configuration File: `ut/host/tests/configs/deviceConfig.yml`
+
+For more details refer [RAFT](https://github.com/rdkcentral/python_raft/blob/1.0.0/README.md) and [example_device_config.yml](https://github.com/rdkcentral/python_raft/blob/1.0.0/examples/configs/example_device_config.yml)
+
+Update the target directory where `HAL` binaries will be copied into the device. Also, map the profile to the source/sink settings `YAML` file path.
+
+Ensure the platform should match with the `DUT` platform in [Rack Configuration](#rack-configuration-file)
+
+```yaml
+deviceConfig:
+  cpe1:
+    platform: "stb"    # Must match the platform in example_rack_config.yml
+    model: "uk"
+    target_directory: "/tmp"  # Path where HAL binaries are copied in device
+    test:
+      profile: "../../../../profiles/source/Source_4K_Display.yaml"
+      player:
+        tool: "gstreamer"
+        prerequisites:
+          - export xxxx    # Pre-commands required to play the stream
+
+```
+
+#### Test Setup Configuration File
```

Comment:
can you check this part in Audio and follow the same

Reply:
Done